### PR TITLE
Make slugs case insensitive

### LINKED ISSUE #247: Make slugs case-insensitive
Right now there's a distinction between `joshsmith` and `JOSHSMITH`, which means someone could ostensibly claim both URLs.

We need to fix this case sensitivity in a way that ensures `JOSHSMITH` resolves to the same URL.

### PR DESCRIPTION
**Pull request by joshsmith**
Closes #247.

### CODE REVIEW

**Comment by joshsmith** [comment]
This could maybe be refactored slightly into a query object, or at least scopes on both `Project` and `SluggedRoute`.

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -111,7 +111,10 @@ def project_show?
 
     def find_project_with_slugged_route!
       slugged_route = find_slugged_route!
-      Project.includes(:github_repositories, :organization).find_by!(slug: project_slug, organization: slugged_route.owner)
+      Project.includes(:github_repositories, :organization).
+        where("lower(slug) = ?", project_slug.downcase).
+        where(organization: slugged_route.owner).
+        first!
```
Comment (begedin) [inline app/controllers/projects_controller.rb]: Would 

``` ruby
where("lower(slug) = ?", project_slug.downcase).
find_by!(organization: slugged_route.owner)
```

not work here?

Comment (joshsmith) [inline app/controllers/projects_controller.rb]: I'm not totally sure. I can try it.

Comment (joshsmith) [inline app/controllers/projects_controller.rb]: It does. Pushing up change.

**Comment by begedin** [comment]
:+1: Looks good to me.

**Pull request merged**